flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java

```diff
@@ -276,6 +280,10 @@ private SqlToOperationConverter(FlinkPlannerImpl flinkPlanner, CatalogManager ca
                     converter.convertBeginStatementSet((SqlBeginStatementSet) validated));
         } else if (validated instanceof SqlEndStatementSet) {
             return Optional.of(converter.convertEndStatementSet((SqlEndStatementSet) validated));
+        } else if (validated instanceof SqlSet) {
```

Comment:
do we need some special handling in the legacy planner? or is the current error behavior helpful?

Reply:
The behavior to return `empty()` seems consistent/sane to me given the contract. If you do end up trying to execute it, you'd get `TableEnvironmentImpl#UNSUPPORTED_QUERY_IN_EXECUTE_SQL_MSG` which is pretty elaborate (and nothing about this changed here anyway – you couldn't run SET through `TableEnvironment#execute*` and you can't now).